gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
+
+        final List<String> properties = new ArrayList<String>();
+
+        for( PropertyKey propertyKey : propertyKeys )
+        {
+            properties.add( propertyKey.getName().content( true ) + "=" + propertyKey.getValue().content( true ) );
+        }
+
         final String projectTemplateName = op.getProjectTemplateName().content();
 
         StringBuilder sb = new StringBuilder();
         sb.append( "create " );
-        sb.append( "-d \"" + location.toFile().getAbsolutePath() +  "\" " );
+        sb.append( "-d \"" + location.toFile().getAbsolutePath() + "\" " );
         sb.append( "-t " + projectTemplateName + " " );
-        sb.append( "\"" + projectName + "\"");
 
-        try
+        if( className != null )
         {
-            BladeCLI.execute( sb.toString() );
+            sb.append( "-c " + className + " " );
         }
-        catch( BladeCLIException e )
+
+        if( serviceName != null )
         {
-            retval = ProjectCore.createErrorStatus( e );
+            sb.append( "-s " + serviceName + " " );
         }
 
-        if( retval.isOK() )
+        if( packageName != null )
         {
-            ProjectImportConfiguration configuration = new ProjectImportConfiguration();
-            GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
-            configuration.setGradleDistribution( from );
-            configuration.setProjectDir( location.toFile() );
-            configuration.setApplyWorkingSets( false );
-            configuration.setWorkingSets( new ArrayList<String>() );
-            new SynchronizeGradleProjectJob(
-                configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
-                AsyncHandler.NO_OP ).schedule();
+            sb.append( "-p " + packageName + " " );
+        }
+
+        sb.append( "\"" + projectName + "\" " );
+
+        try
+        {
+            final String[] ret = BladeCLI.execute( sb.toString() );
+
+            final String errors = checkForErrors(ret);
+
+            if( errors.length() > 0 )
+            {
+                retval = GradleCore.createErrorStatus( "Project create error: " + errors );
+                return retval;
+            }
+
+            if( ret.length == 0 )
+            {
+                ElementList<ProjectName> projectNames = op.getProjectNames();
+                
+                projectNames.insert().setName( projectName );
+                
+                IPath projecLocation = location;
+
+                final String lastSegment = location.lastSegment();
+
+                if( location != null && location.segmentCount() > 0 )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
```

Comment:
Coordinate with Andy.  he has developed a workspace wizard that imports a gradle project.  He created a GradleUtil for doing this import.  Go ahead and add the util method to GradleUtil as a part of this pull request.